### DISCUSSION TITLE
Consul lock prefix does not match cluster_formation.consul.lock_prefix

### DISCUSSION THREAD

**Discussion by xixka** [rabbitmq/rabbitmq-server#2700] [Other] [answered]
Dockerfile
```
FROM rabbitmq:3.8.9-management
COPY erlang.cookie /var/lib/rabbitmq/.erlang.cookie
COPY rabbitmq.conf /etc/rabbitmq/rabbitmq.conf
RUN chmod 777 /etc/rabbitmq/rabbitmq.conf
ENV RABBITMQ_VM_MEMORY_HIGH_WATERMARK=0.9
RUN rabbitmq-plugins --offline enable rabbitmq_peer_discovery_consul
```
rabbitmq.conf
```
cluster_formation.peer_discovery_backend = consul
cluster_formation.consul.host = consul-1
cluster_formation.consul.svc_addr_auto = true
cluster_formation.consul.deregister_after = 90
cluster_formation.consul.lock_prefix = ir
```
Startup parameters
```
    environment:
      - RABBITMQ_DEFAULT_USER=ir
      - RABBITMQ_DEFAULT_PASS=ir
      - RABBITMQ_NODENAME=rabbitmq
```
Then the following picture bug appears
lock_prefix name is **irloopback_users.guest = false**
![屏幕截图 2020-12-29 171502](https://user-images.githubusercontent.com/73456104/103273221-b490b980-49f9-11eb-820b-ab0f392dd5e7.png)

**Comment by michaelklishin** [comment]
There is no evidence of a bug in RabbitMQ. The Consul plugin configuration schema generation is [reasonably thoroughly tested](https://github.com/rabbitmq/rabbitmq-server/blob/master/deps/rabbitmq_peer_discovery_consul/test/config_schema_SUITE_data/rabbitmq_peer_discovery_consul.snippets) so I don't expect any surprises there.

I remember seeing something like this before and it was a Docker image configuration "generation" side effect IIRC. The image concatenates its `loopback_users` setting without correctly adding a new line, so final `rabbitmq.conf` ends up begin

``` ini
# note the missing new line
cluster_formation.consul.lock_prefix = irloopback_users.guest = false
```

or something like this. You can [inspect effective node configuration](https://www.rabbitmq.com/configure.html#verify-configuration-effective-configuration) to further confirm this. You can also try adding extra new lines at the end.

**Reply by michaelklishin** [reply]
[Here is where](https://github.com/docker-library/rabbitmq/blob/master/3.8/ubuntu/docker-entrypoint.sh#L298) the Docker image messes with the configuration file (for historical reasons).

If you unset `RABBITMQ_DEFAULT_USER` and `RABBITMQ_DEFAULT_PASS` , I suspect it won't mess with `rabbitmq.conf`.
Our team strongly recommends **against** using environment variables where they can be avoided, with Docker or not.
Sensitive configuration values such as default user credentials can be [encrypted in `advanced.config`](https://www.rabbitmq.com/configure.html#configuration-encryption).

**Comment by michaelklishin** [comment, marked as answer]
Filed https://github.com/docker-library/rabbitmq/issues/456 as I cannot reproduce this with just a `rabbitmq.conf` and a Consul node:

``` ini
cluster_formation.peer_discovery_backend = rabbitmq_peer_discovery_consul

cluster_formation.consul.host = localhost

cluster_formation.consul.svc = rabbitmq

## do compute service address
# cluster_formation.consul.svc_addr_auto = true
## compute service address using node name
# cluster_formation.consul.svc_addr_use_nodename = true
## use long RabbitMQ node names?
# cluster_formation.consul.use_longname = true

cluster_formation.consul.svc_addr_auto = true
# cluster_formation.consul.svc_addr = warp10
cluster_formation.consul.use_longname = true

# cluster_formation.proxy.http_proxy = proxy.example.local
# cluster_formation.proxy.proxy_exclusions.1 = localhost
# cluster_formation.proxy.proxy_exclusions.2 = 127.0.0.1

cluster_formation.consul.svc_tags.1 = qa
cluster_formation.consul.svc_tags.2 = 3.8

cluster_formation.consul.svc_meta.owner = team-xyz
cluster_formation.consul.svc_meta.service = service-one
cluster_formation.consul.svc_meta.stats_url = https://service-one.eng.megacorp.local/stats/

# cluster_formation.consul.svc_ttl = 60

log.console.level = debug
log.file.level = debug

cluster_formation.consul.lock_prefix = rabbitmq-prefix
```

The prefix is what I'd expect:

![Screenshot 2020-12-29 at 15 19 27](https://user-images.githubusercontent.com/1090/103283302-447a3780-49e9-11eb-8eab-7bff35b0b35c.png)
